src/activity_handlers.cpp

```diff
@@ -1120,18 +1121,18 @@ void serialize_liquid_target( player_activity &act, const monster &mon )
     act.coords.push_back( mon.pos() );
 }
 
-void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
```

Comment:
Larger issue, is activity ever null here?  If not it should just be a ref in the first place.

Reply:
All activity functions (i.e. `_do_turn` and `_finish`) have same parameters for consistency so they could be called from `void activity_type::call_do_turn( player_activity *act, player *p ) const`, so I can't change parameter to reference in onyl one of the functions. I didn't see any null checks in activities though, so we can probably later refactor this functions to use references.

Reply:
To clarify, by "larger issue" I mean I'm not asking to handle it in this issue, it's already quite big enough ^_^

Yea I suspect we should do that refactor, because we are certainly acting like there's a player there, and if there isn't, we need to not call the function at all.